docs/PAYLOAD.md

```diff
@@ -72,6 +72,121 @@ Some ways to handle this *double* event are:
+    "notId": 1, // unique ID for the message, used for grouping, see below
+    "custom_key1": "value1",
+    "custom_key2": "value2"
+  },
+  // "notification": { ... } // can be used instead of data for title and message, see ....
+}
+```
+
+This is the JSON-encoded format you can e.g. send via AWS-SNS's web UI.
+Note, that the core message is json-encoded twice, so if we take the `content` from above you convert it this way
+
+```javascript
+  var gcm_message = JSON.stringify({
+    "GCM": JSON.stringify(content),
+    "default": "plain text message again"
+  });
+```
+
+```json
+{"GCM":
+  "{\"priority\":\"normal\",\"content_available\":\"0\",\"data\":{\"title\":\"A short string describing the purpose of the notification\",\"message\":\"The text of the alert message\",\"count\":5,\"sound\": \"default\",\"notId\":1,\"custom_key1\":\"value1\",\"custom_key2\":\"value2\"}}"
+}
+```
+
+This message is received in the "notification" handler as follows.
+Note that the properties are "normalized" across platforms, so this is passed to the app on android:
+
+```json
+{
+  "count": "5",
+  "message": "The text of the alert message",
+  "sound": "default",
+  "title": "A short string describing the purpose of the notification",
+  "additionalData": {
+    "custom_key1": "value1",
+    "custom_key2": "value2",
+    "notId": "1",
+    "dismissed": false,
+    "google.message_id": "...",
+    "coldstart": false,
+    "foreground": false
+  }
+}
+```
+
+## iOS Message Format
+
+The JSON message can contain the following fields, see [Apple developer docs](https://developer.apple.com/library/content/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/PayloadKeyReference.html#//apple_ref/doc/uid/TP40008194-CH17-SW5) for a complete list
+
+```json
+{
+  "aps": {
+    "alert": { // alternatively just a string: "Your Message",
+      "title": "A short string describing the purpose of the notification",
+      "body": "The text of the alert message",
+      // localozation of message is possible
+      "launch-image": "The filename of an image file in the app bundle, with or without the filename extension. The image is used as the launch image when users tap the action button or move the action slider"
+    },
+    "badge": 5, // Number to show at App icon
+    "content-available": 0, // configure background updates, see below
+    "category": "identifier", // Provide this key with a string value that represents the notification’s type
+    "thread-id": "id", // Provide this key with a string value that represents the app-specific identifier for grouping notifications
+    "sound": "default"  // play default sound, or custom sound, see [iOS Sound](#sound-1) section
+  },
+  "custom_key1": "value1",
+  "custom_key2": "value2"
+}
+```
+
+This is the JSON-encoded format you can send via AWS-SNS's web UI (use "APNS" or "APNS_SANDBOX" depending on which version you use to publish):
```

Comment:
`(use "APNS" to send to an app signed and released to production or "APNS_SANDBOX" or to send to an app signed and released for development)`